From 2539dd09b2f5efba5f548cfdcf573c842fc72196 Mon Sep 17 00:00:00 2001
From: Jianxun Zhang <jianxun.zhang@intel.com>
Date: Mon, 25 Nov 2013 12:08:19 -0800
Subject: Fix dependency of modsign keys

In the kernel the auto-generated key pair depends on the genkey
file in Makefile.

We need to ensure the transformed key pair newer than the copied
genkey file before building the kernel so the key pair will not
be over-written by the auto-generated keys that are not enrolled
in BIOS for module validation.

Issue: APDEV-499
Change-Id: I6b9b7749b61d5816cce03994c54523e5ec7506df
Signed-off-by: Jianxun Zhang <jianxun.zhang@intel.com>
---
 core/kernel.mk |    8 +++-----
 1 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/core/kernel.mk b/core/kernel.mk
index d0b0332..a1cb518 100644
--- a/core/kernel.mk
+++ b/core/kernel.mk
@@ -118,14 +118,12 @@ kernel_signed_mod_enabled = $(shell grep ^CONFIG_MODULE_SIG=y $(kernel_config_fi
 kernel_genkey := $(kbuild_output)/x509.genkey
 kernel_private_key := $(kbuild_output)/signing_key.priv
 kernel_public_key := $(kbuild_output)/signing_key.x509
-kernel_key_deps := $(if kernel_signed_mod_enabled,$(kernel_genkey) $(kernel_private_key) $(kernel_public_key))
+kernel_key_deps := $(if kernel_signed_mod_enabled,$(kernel_private_key) $(kernel_public_key))
 
-$(kernel_public_key): $(TARGET_MODULE_KEY_PAIR).x509.pem
-	$(hide) mkdir -p $(dir $@)
+$(kernel_public_key): $(TARGET_MODULE_KEY_PAIR).x509.pem $(kernel_genkey)
 	$(hide) openssl x509 -inform PEM -outform DER -in $(TARGET_MODULE_KEY_PAIR).x509.pem -out $@
 
-$(kernel_private_key): $(TARGET_MODULE_KEY_PAIR).pk8
-	$(hide) mkdir -p $(dir $@)
+$(kernel_private_key): $(TARGET_MODULE_KEY_PAIR).pk8 $(kernel_genkey)
 	$(hide) openssl pkcs8 -nocrypt -inform DER -outform PEM -in $(TARGET_MODULE_KEY_PAIR).pk8 -out $@
 
 $(kernel_genkey): $(TARGET_MODULE_GENKEY) | $(ACP)
-- 
1.7.4.1

